File: 658-find-k-closest-elements/find-k-closest-elements.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        for (int i = 0; i < arr.size(); i++) {
            int diff = abs(x - arr[i]);
            int idx = i;
            for (int j = i + 1; j < arr.size(); j++) {
                int dif = abs(x - arr[j]);
                // cout << dif << " " ;
                if (dif < diff) {
                    // cout << dif << " " << arr[j] << dif << endl;
                    idx = j;
                    diff = dif;
                }
            }
            // cout << arr[idx] << endl;
            swap(arr[i], arr[idx]);
        }
        vector<int>result;
        sort(arr.begin(), arr.begin() + k);
        for(int i = 0 ; i < k ;i++){
            result.push_back(arr[i]);
        }
        return result;
    }
};

File: 658-find-k-closest-elements/find-k-closest-elements.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {

        int i = 0;
        int j = arr.length - 1;


        while ((j - i + 1) != k) {

            int a = Math.abs(x - arr[i]);
            int b = Math.abs(x - arr[j]);

            if (a > b) {
                i++;
            } else {
                j--;
            }
        }

        List<Integer> list = new ArrayList<>();

        for (int it = i; it <= j; it++) {
            list.add(arr[it]);
        }

        return list ; 

    }
}